packages/main/src/plugin/api/pod-info.ts

```diff
@@ -29,6 +29,9 @@ export interface PodInfo extends LibPodPodInfo {
   engineId: string;
   engineName: string;
   kind: 'kubernetes' | 'podman';
+  // Used for Kubernetes
+  node?: string;
+  namespace?: string;
```

Comment:
Instead of adding a lot of optional fields, I would rather have a new interface

```ts
interface KubernetesPodInfo extend PodInfo {
    node: string;
    namespace: string;
}
```
And an utility method

```ts
export function isKubernetesPod(pod: PodInfo): pod is KubernetesPodInfo  {
    return pod.kind === 'kubernetes';
}
```